pages/api/v1/universities/index.ts

```diff
@@ -3,36 +3,80 @@ import fs from "fs";
 import path from "path";
 import { University } from "@/types/university.type";
 
+let cachedUniversities: University[] | null = null;
+
+function loadUniversities(): University[] {
```

Comment:
walk me through why you have this @stephengade.

Reply:
`let cachedUniversities: University[] | null = null` 

This is just to store universities' data once it's loaded - `null`, the initial stage when the data hasn't been loaded

`loadUniversities`  function that will load the universities' data. before it loads the data, it should first check if the data has already been cached in cachedUniversities

If the data is already cached, the function can return it directly without needing to read it all over again

PS: Better caching mechanism can be used

Reply:
we'll figure out a better caching approach as we progress. This should work fine for now. 

Thanks again!